build_jar.py

```diff
@@ -95,7 +97,7 @@
             args.source,
             "konduit-serving-uberjar",
             "target",
-            "konduit-serving-uberjar-{}-bin.jar".format(version[0]),
+            "konduit-serving-uberjar-{}-custom-{}-{}.jar".format(version[0], args.os, arch),
```

Comment:
@philip-khor Just for reference, the idea behind a spin version is that we want to have multiple iterations of the built uber jar. For example, if I have `spin.version=minimal` it will build the jar file without the `pmml` and `python` profiles. Similarly, the other types are:

- `spin.version=python` -> add the `python` profile
- `spin.version=pmml` -> add the `pmml` profile 
- `spin.version=all` (keep this as default in `build_jar.py`) -> add `python` and `pmml` profiles

Based on that can you make changes to add an additional `spin` argument that takes care of those profiles internally and additionally add a `-Dspin.version=<spin>` to the build command? 

I see that we have `usePmml` and `usePython` in there as well, you might wanna remove them in order to make this workflow and update documentation as well.

Reply:
Changes made. I realised CLI (`konduit build` and `konduit init`) need that too. I can go ahead and try making those changes there too.